ocaml/database/redo_log.ml

```diff
@@ -378,7 +378,7 @@ let action_write_db marker generation_count write_fn sock datasockpath =
          write_fn datasock;
          R.debug "Finished writing database to data socket";
        with
-       | Sys_error("Connection reset by peer") ->
+       | Sys_error e when e = "Connection reset by peer" ->
```

Comment:
This gets rid of the issue, but the fragility from depending on the particular string values is still present.

Not sure how to solve this

Reply:
Unfortunately no unless you drop the use of any function raising Sys_error. This is why we left them as they were. You could silence the warnings with `[@warning "-X"]` where X is the warning number

Reply:
Similarly in cli_operations